modules/controller/world_model_tracking.py

```diff
@@ -0,0 +1,86 @@
+        self.__false_positives = []
+        self.confirmed_positive = None
+
+    def __similar(self,
+                  detection1: object_in_world.ObjectInWorld,
+                  detection2: object_in_world.ObjectInWorld) \
+                -> bool:
+        """
+        returns whether detection1 and detection2 are close enough
+        to be considered the same landing pad
+        """
+
+        DISTANCE_THRESHOLD = 2
+        distance = (detection2.position_x - detection1.position_x) ** 2 \
+                 + (detection2.position_y - detection1.position_y) ** 2
+        return distance < DISTANCE_THRESHOLD
+
+    def mark_false_positive(self, detection: object_in_world.ObjectInWorld):
+        """
+        marks a detection as false positive and removes similar landing
+        pads from the list of unconfirmed positives
+        """
+
+        self.__false_positives += [detection]
+        for landing_pad in self.__unconfirmed_positives:
+            if self.__similar(landing_pad, detection):
+                self.__unconfirmed_positives.remove(landing_pad)
+                break
+        return detection
+
+    def mark_confirmed_positive(self, detection: object_in_world.ObjectInWorld) \
+                                -> object_in_world.ObjectInWorld:
+        """
+        marks a detection as a confimred positive for future use
+        """
+
+        self.confirmed_positive = detection
+        return self.confirmed_positive
+
+    def run(self, detections: np.ndarray):
+        """
+        updates the list of unconfirmed positives and returns the
+        detection with the lowest variance
+        """
+
+        for detection in detections:
+            match_found = False
+            # if detection matches a false positive, don't add it
+            for false_positive in self.__false_positives:
+                if self.__similar(detection, false_positive):
+                    match_found = True
+                    break
+            if match_found:
+                continue
+            # if detection matches an unconfirmed positive, replace old detection
+            for i, landing_pad in enumerate(self.__unconfirmed_positives):
+                if self.__similar(detection, landing_pad):
+                    match_found = True
+                    self.__unconfirmed_positives[i] = detection
+                    break
+            if match_found:
+                continue
+            # if new landing pad, add to list of unconfirmed positives
+            self.__unconfirmed_positives.append(detection)
+
+        # if the list is empty, all landing pads have been visited, none are viable
+        if len(self.__unconfirmed_positives) == 0:
+            return False, None
+        
+        # sort list by variance
+        self.__unconfirmed_positives.sort(key=lambda x: x.spherical_variance)
+        # return detection with lowest variance
+        return True, self.__unconfirmed_positives[0]
```

Comment:
What about the case where there are confirmed landing pads? We would want to land there instead.

Reply:
The run function returns the detection with the lowest variance (since self.__unconfirmed_positives is sorted), so it would land there.

Reply:
The goal of the class is to return the best landing pad to land on. If there is a confirmed landing pad, it would be preferred over any of the unconfirmed landing pads. Otherwise, the landing pad with the lowest variance would be the best.

Reply:
If there are multiple confirmed landing pads, which should it return?

Reply:
The first index should be fine.